tests/common/widgets/search-box/options.ts

```diff
@@ -0,0 +1,461 @@
+        'Search here'
+      );
+    });
+
+    test('refines query when typing by default', async () => {
+      let state: UiState = {};
+      const searchClient = createSearchClient({});
+
+      await setup({
+        instantSearchOptions: {
+          indexName: 'indexName',
+          searchClient,
+          onStateChange: ({ uiState }) => (state = uiState),
+        },
+        widgetParams: {},
+      });
+
+      await act(async () => {
+        await wait(0);
+      });
+
+      userEvent.type(screen.getByRole('searchbox'), 'iPhone');
+
+      expect(state.indexName.query).toBe('iPhone');
+      expect(screen.getByRole('searchbox')).toHaveValue('iPhone');
+    });
+
+    test('refines when hitting the Enter key', async () => {
+      let state: UiState = {};
+      const searchClient = createSearchClient({});
+
+      await setup({
+        instantSearchOptions: {
+          indexName: 'indexName',
+          searchClient,
+          onStateChange: ({ uiState }) => (state = uiState),
+        },
+        widgetParams: { searchAsYouType: false },
+      });
+
+      await act(async () => {
+        await wait(0);
+      });
+
+      userEvent.type(screen.getByRole('searchbox'), 'iPhone');
+
+      await act(async () => {
+        await wait(0);
+      });
+
+      userEvent.type(screen.getByRole('searchbox'), '{enter}');
+
+      expect(state.indexName.query).toBe('iPhone');
+    });
+
+    test('refines when clicking on the submit button', async () => {
+      let state: UiState = {};
+      const searchClient = createSearchClient({});
+
+      await setup({
+        instantSearchOptions: {
+          indexName: 'indexName',
+          searchClient,
+          onStateChange: ({ uiState }) => (state = uiState),
+        },
+        widgetParams: { searchAsYouType: false },
+      });
+
+      await act(async () => {
+        await wait(0);
+      });
+
+      userEvent.type(screen.getByRole('searchbox'), 'iPhone');
+
+      await act(async () => {
+        await wait(0);
+      });
+
+      userEvent.click(
+        screen.getByRole('button', {
+          name: /submit the search query/i,
+        })
+      );
+
+      expect(state.indexName.query).toBe('iPhone');
+    });
+
+    test('resets query when clicking on reset button', async () => {
+      let state: UiState = {};
+      const searchClient = createSearchClient({});
+
+      await setup({
+        instantSearchOptions: {
+          indexName: 'indexName',
+          searchClient,
+          onStateChange: ({ uiState }) => (state = uiState),
+          initialUiState: {
+            indexName: {
+              query: 'something',
+            },
+          },
+        },
+        widgetParams: {},
+      });
+
+      await act(async () => {
+        await wait(0);
+      });
+
+      userEvent.click(
+        screen.getByRole('button', {
+          name: /clear the search query/i,
+        })
+      );
+
+      expect(state.indexName.query).not.toBeDefined();
```

Comment:
here and in the other tests which test the state, we also need to test that the input value is the one you expect

(I know the dashboard had a bug until last week where clear didn't actually clear the input value, which is easy to accidentally be in because of the state alias)

Reply:
There's an issue with Vue in a testing environment where the searchbox still reports the previous value, not sure how to fix it. Do you have an idea?

Reply:
Found it: we took over `onStateChange` and did not propagate the new ui state ourselves.